scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -86,4 +86,9 @@ private[scio] object ScioUtil {
 
   def pathWithShards(path: String): String =
     path.replaceAll("\\/+$", "") + "/part"
+
+  def hashCodeFn[K]: K => Int = {
```

Comment:
we should make this ` def hashCodeFn[K](k :K): Int`. I feel like we should change the name as well, perhaps smth like `consistentHashCode`?